### PR TITLE
Small clean to SymmetricalLogLocator

### PR DESCRIPTION
**Pull request by dstansby**
- Replace variable names `b, t` with `base, linthresh`
- Add some more inline comments
- Instead of doing `-vmin, -vmax`, use `np.abs()`, as the intent is to make sure the value passed to `get_log_range()` is positive.

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -2492,46 +2493,47 @@ def tick_values(self, vmin, vmax):
...
+            lo = np.floor(np.log(lo) / np.log(base))
+            hi = np.ceil(np.log(hi) / np.log(base))
             return lo, hi
 
-        # First, calculate all the ranges, so we can determine striding
+        # Calculate all the ranges, so we can determine striding
         if has_a:
             if has_b:
-                a_range = get_log_range(t, -vmin + 1)
+                a_range = get_log_range(linthresh, np.abs(vmin))
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py]: Why no `+ 1`?

File: lib/matplotlib/ticker.py [hunk 2]
```diff
@@ -2492,46 +2493,47 @@ def tick_values(self, vmin, vmax):
...
             else:
-                a_range = get_log_range(-vmax, -vmin + 1)
+                a_range = get_log_range(np.abs(vmax), np.abs(vmin))
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py]: Why no `+ 1`?

File: lib/matplotlib/ticker.py (hunk 1 above)
Comment (dstansby) [reply lib/matplotlib/ticker.py]: Woops, that accidentally got lost

File: lib/matplotlib/ticker.py [hunk 3]
```diff
@@ -2468,22 +2468,23 @@ def tick_values(self, vmin, vmax):
         # "simple" mode is when the range falls entirely within (-t,
         # t) -- it should just display (vmin, 0, vmax)
 
+        # Determine which of the three ranges we have
```
Comment (anntzer) [inline lib/matplotlib/ticker.py]: I think the whole thing can be simplified to
```
if -linthresh <= vmin <= vmax <= 0 or 0 <= vmin <= vmax <= linthresh:
    return [vmin, vmax]
has_a = (vmin < -linthresh)
has_b = (-linthresh < vmin < linthresh or -linthresh < vmax < linthresh)
has_c = (linthresh < vmax)
```

File: lib/matplotlib/ticker.py [hunk 4]
```diff
@@ -2492,46 +2493,47 @@ def tick_values(self, vmin, vmax):
...
         if has_c:
             if has_b:
-                c_range = get_log_range(t, vmax + 1)
+                c_range = get_log_range(linthresh, vmax + 1)
             else:
                 c_range = get_log_range(vmin, vmax + 1)
         else:
             c_range = (0, 0)
 
+        # Caclulate the total number of integer exponents in a and c ranges
```
Comment (anntzer) [inline lib/matplotlib/ticker.py]: typo: calculate

File: lib/matplotlib/ticker.py (hunk 3 above)
Comment (dstansby) [reply lib/matplotlib/ticker.py]: I think `SymLogNorm` needs quite a few more tests, which I plan to add in for `3.2` at some point; I think this is a good suggestion, but might save it for after I've written tests that exercise this logic properly.

Comment (anntzer) [reply lib/matplotlib/ticker.py]: sure

File: lib/matplotlib/ticker.py [hunk 5]
```diff
@@ -2492,46 +2493,47 @@ def tick_values(self, vmin, vmax):
...
+            lo = np.floor(np.log(lo) / np.log(base))
+            hi = np.ceil(np.log(hi) / np.log(base))
             return lo, hi
 
-        # First, calculate all the ranges, so we can determine striding
+        # Calculate all the ranges, so we can determine striding
         if has_a:
             if has_b:
-                a_range = get_log_range(t, -vmin + 1)
+                a_range = get_log_range(linthresh, np.abs(vmin) + 1)
```
Comment (anntzer) [inline lib/matplotlib/ticker.py]: can use (builtin) abs() instead of np.abs()

**Comment by dstansby** [comment]
Going to merge, since this has two approvals.

**Pull request merged**